### PR TITLE
Add client authentication support for ticdc

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: ticdc/ticdc-client-authentication.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+title: TiCDC 客户端鉴权
+summary: 了解 TiCDC cli 和 api 的鉴权方式。
+---
+
+# TiCDC 客户端鉴权
+
+> **注意：**
+>
+> TiCDC 只支持在开启 tls 加密时进行客户端鉴权。
```
Comment (flowbehappy) [inline ticdc/ticdc-client-authentication.md:10]: 描述一下下面两种鉴权方式是否可以共存、单独使用的情况

File: ticdc/ticdc-client-authentication.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+## 通过 mtls 进行客户端鉴权
```
Comment (flowbehappy) [inline ticdc/ticdc-client-authentication.md:12]: 可以描述下这种鉴权方式的意义

File: ticdc/ticdc-client-authentication.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+- 使用 cli 时，cdc 按照以下顺序读取客户端证书：
+  1. 通过命令行参数 `--cert` 和 `--key` 指定用于鉴权的证书和私钥。如果服务端使用了自签名证书，还需要通过 `--ca` 指定受信任的 CA 证书。
+  2. 通过环境变量 `TICDC_CA_PATH`、`TICDC_CERT_PATH`、`TICDC_KEY_PATH` 配置客户端证书。
+  3. 通过共享凭证文件 `~/.ticdc/credentials` 获取客户端证书。你可以通过命令 `cdc cli configure-credentials` 更改这个文件的配置。
+
+- 使用 curl 命令访问 api 时：
+  - 通过 `--cert` 和 `--key` 指定客户端证书。如果服务端使用了自签名证书，还需要通过 `--cacert` 指定受信任的 CA 证书。
+
+
+## 通过 TiDB 用户进行客户端鉴权
```
Comment (flowbehappy) [inline ticdc/ticdc-client-authentication.md:30]: 描述下这种鉴权方式的意义

Comment (flowbehappy) [reply ticdc/ticdc-client-authentication.md:30]: 描述 user 和 tidb 中的 user 有什么关系，需要在 tidb 中提前添加用户，以及这些用户在 tidb 中需要什么权限

File: ticdc/ticdc-client-authentication.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+title: TiCDC 客户端鉴权
+summary: 了解 TiCDC cli 和 api 的鉴权方式。
+---
+
+# TiCDC 客户端鉴权
+
+从 v8.1.0 起，TiCDC 支持通过 mtls 或 TiDB 用户名密码进行客户端鉴权：
+  1. mtls（双向传输层安全协议）鉴权：提供了在传输层进行安全控制的方法，使 TiCDC 可以验证客户端身份。
+  2. 通过 TiDB 用户名密码鉴权：提供了在应用层进行安全控制的方法，合法用户需要有通过 TiCDC 节点进行登录的权限。
+这些方式可以单独使用，也可以结合起来使用，以满足不同的场景和安全需求。
```
Comment (CharlesCheung96) [inline ticdc/ticdc-client-authentication.md:11]:
```suggestion
这些鉴权方式可以单独使用，也可以结合起来使用，以满足不同的场景和安全需求。
```

File: ticdc/ticdc-client-authentication.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
+---
+title: TiCDC 客户端鉴权
+summary: 了解 TiCDC cli 和 api 的鉴权方式。
+---
+
+# TiCDC 客户端鉴权
+
+从 v8.1.0 起，TiCDC 支持通过 mtls 或 TiDB 用户名密码进行客户端鉴权：
```
Comment (CharlesCheung96) [inline ticdc/ticdc-client-authentication.md:8]:
```suggestion
从 v8.1.0 起，TiCDC 支持通过 mtls 或 TiDB 用户名密码进行客户端鉴权。
```

**Review by flowbehappy — APPROVED** [review]

File: ticdc/ticdc-client-authentication.md
Comment (Oreoxmt) [inline ticdc/ticdc-client-authentication.md:1]: Preview: https://pingcap-docsite-preview-git-preview-pi-2ab57e-oreoxmts-projects.vercel.app/zh/tidb/dev/ticdc-client-authentication

**Review by Oreoxmt — APPROVED** [review]

File: ticdc/ticdc-client-authentication.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+title: TiCDC 客户端鉴权
+summary: 介绍使用 TiCDC 命令行工具或通过 OpenAPI 访问 TiCDC 时，如何进行客户端鉴权。
+---
+
+# TiCDC 客户端鉴权
+
+从 v8.1.0 起，TiCDC 支持使用 mTLS（双向传输层安全性协议）或 TiDB 用户名密码进行客户端鉴权。
+
+- mTLS 鉴权：提供了在传输层进行安全控制的方法，使 TiCDC 可以验证客户端身份。
+- TiDB 用户名密码鉴权：提供了在应用层进行安全控制的方法，确保只有授权用户才能通过 TiCDC 节点登录。
```
Comment (qiancai) [inline ticdc/ticdc-client-authentication.md:11]:
```suggestion
- mTLS 鉴权：在传输层进行安全控制，使 TiCDC 可以验证客户端身份。
- TiDB 用户名密码鉴权：在应用层进行安全控制，确保只有授权用户才能通过 TiCDC 节点登录。
```

File: ticdc/ticdc-client-authentication.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+        export TICDC_CA_PATH=ca.crt
+        ```
+
+    3. 通过共享凭证文件 `~/.ticdc/credentials` 指定客户端证书。你可以使用 `cdc cli configure-credentials` 命令修改此文件的配置。
+
+    </div>
+
+    <div label="TiCDC OpenAPI" value="cdc-api">
+
+    使用 [TiCDC OpenAPI](/ticdc/ticdc-open-api-v2.md) 时，通过 `--cert` 和 `--key` 指定客户端证书。如果服务端使用了自签名证书，还需要通过 `--cacert` 指定受信任的 CA 证书。示例：
```
Comment (qiancai) [inline ticdc/ticdc-client-authentication.md:56]:
```suggestion
    使用 [TiCDC OpenAPI](/ticdc/ticdc-open-api-v2.md) 时，通过 `--cert` 和 `--key` 指定客户端证书和私钥。如果服务端使用了自签名证书，还需要通过 `--cacert` 指定受信任的 CA 证书。示例：
```

File: ticdc/ticdc-client-authentication.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+
+    ```bash
+    curl -X GET http://127.0.0.1:8300/api/v2/status --cert client.crt --key client.key --cacert ca.crt
+    ```
+
+    </div>
+    </SimpleTab>
+
+## 使用 TiDB 用户进行客户端鉴权
```
Comment (qiancai) [inline ticdc/ticdc-client-authentication.md:65]:
```suggestion
## 使用 TiDB 用户名密码进行客户端鉴权
```

File: ticdc/ticdc-client-authentication.md [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+    # 指定可用于客户端鉴权的用户名，列表中不存在的鉴权请求将被直接拒绝。默认值为 null。
+    client-allowed-user = ["test"]
+    ```
+
+3. 指定步骤 1 创建的授权用户的用户名和密码：
+
+    <SimpleTab groupId="cdc">
+    <div label="TiCDC 命令行工具" value="cdc-cli">
+
+    使用 [TiCDC 命令行工具](/ticdc/ticdc-manage-changefeed.md)时，你可以通过以下方式指定用户名和密码，TiCDC 将按照以下顺序依次尝试读取用户名和密码：
```
Comment (qiancai) [inline ticdc/ticdc-client-authentication.md:88]:
```suggestion
    使用 [TiCDC 命令行工具](/ticdc/ticdc-manage-changefeed.md)时，你可以通过以下方式之一指定用户名和密码。TiCDC 将按照以下顺序依次尝试读取用户名和密码：
```

File: ticdc/ticdc-client-authentication.md [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+    CREATE USER 'test'@'ticdc_ip_address' IDENTIFIED BY 'password';
+    ```
+
+2. 在 TiCDC Server 中，配置 `security.client-user-required` 和 `security.client-allowed-user` 以开启用户名和密码鉴权：
+
+    ```toml
+    [security]
+    # 控制是否使用用户名和密码进行客户端鉴权，默认值为 false。
+    client-user-required = true
+    # 指定可用于客户端鉴权的用户名，列表中不存在的鉴权请求将被直接拒绝。默认值为 null。
```
Comment (qiancai) [inline ticdc/ticdc-client-authentication.md:79]:
```suggestion
    # 指定可用于客户端鉴权的用户名，列表中不存在的用户的鉴权请求将被直接拒绝。默认值为 null。
```

File: ticdc/ticdc-server-config.md [hunk 11]
```diff
@@ -47,6 +47,12 @@ gc-tuner-memory-threshold = 0
   ca-path = ""
   cert-path = ""
   key-path = ""
+  # 控制是否开启 TLS 客户端鉴权，默认值为 false。
+  mtls = false
+  # 控制是否使用用户名和密码进行客户端鉴权，默认值为 false。
+  client-user-required = false
+  # 指定可用于客户端鉴权的用户名，列表中不存在的鉴权请求将被直接拒绝。默认值为 null。
```
Comment (qiancai) [inline ticdc/ticdc-server-config.md:54]:
```suggestion
  # 指定可用于客户端鉴权的用户名，列表中不存在的用户的鉴权请求将被直接拒绝。默认值为 null。
```

**Review by qiancai — COMMENTED** [review]
Rest LGTM

File: ticdc/ticdc-client-authentication.md [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+
+    ```toml
+    [security]
+    # 控制是否使用用户名和密码进行客户端鉴权，默认值为 false。
+    client-user-required = true
+    # 指定可用于客户端鉴权的用户名，列表中不存在的用户的鉴权请求将被直接拒绝。默认值为 null。
+    client-allowed-user = ["test"]
+    ```
+
+3. 指定步骤 1 创建的授权用户的用户名和密码：
```
Comment (qiancai) [inline ticdc/ticdc-client-authentication.md:83]:
```suggestion
3. 指定步骤 1 创建的授权用户的用户名和密码。
```

File: ticdc/ticdc-client-authentication.md [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+从 v8.1.0 起，TiCDC 支持使用 mTLS（双向传输层安全性协议）或 TiDB 用户名密码进行客户端鉴权。
+
+- mTLS 鉴权：在传输层进行安全控制，使 TiCDC 可以验证客户端身份。
+- TiDB 用户名密码鉴权：在应用层进行安全控制，确保只有授权用户才能通过 TiCDC 节点登录。
+
+这两种鉴权方式既可以单独使用，也可以结合使用，以满足不同的场景和安全需求。
+
+> **注意：**
+>
+> 生产环境中仅建议在[开启 TLS 加密传输](/enable-tls-between-clients-and-servers.md)的情况下，使用 TiCDC 客户端鉴权功能。
```
Comment (Oreoxmt) [inline ticdc/ticdc-client-authentication.md:17]:
```suggestion
> 在生产环境中，建议仅在[开启 TLS 加密传输](/enable-tls-between-clients-and-servers.md)的情况下使用 TiCDC 客户端鉴权功能。
```

File: ticdc/ticdc-client-authentication.md [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+从 v8.1.0 起，TiCDC 支持使用 mTLS（双向传输层安全性协议）或 TiDB 用户名密码进行客户端鉴权。
+
+- mTLS 鉴权：在传输层进行安全控制，使 TiCDC 可以验证客户端身份。
+- TiDB 用户名密码鉴权：在应用层进行安全控制，确保只有授权用户才能通过 TiCDC 节点登录。
+
+这两种鉴权方式既可以单独使用，也可以结合使用，以满足不同的场景和安全需求。
+
+> **注意：**
+>
+> 为了保证网络访问的安全性，建议仅在[开启 TLS 加密传输](/enable-tls-between-clients-and-servers.md)的情况下，使用 TiCDC 客户端鉴权功能。
```
Comment (flowbehappy) [inline ticdc/ticdc-client-authentication.md:17]:
```suggestion
> 为了保证网络访问的安全性，强烈建议仅在[开启 TLS 加密传输](/enable-tls-between-clients-and-servers.md)的情况下，使用 TiCDC 客户端鉴权功能。如果不开启 TLS 加密传输，用户名和密码将会以明文的方式通过网络传输，这会带来严重的泄露风险。
```

**Review by qiancai — APPROVED** [review]

File: ticdc/ticdc-client-authentication.md [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+
+1. 在 TiCDC Server 中，将 `security.mtls` 配置为 `true` 以开启 mTLS 鉴权：
+
+    ```toml
+    [security]
+    # 控制是否开启 TLS 客户端鉴权，默认值为 false。
+    mtls = true
+    ```
+
+2. 配置客户端证书：
```
Comment (qiancai) [inline ticdc/ticdc-client-authentication.md:29]:
```suggestion
2. 配置客户端证书。
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Pull request merged**